test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -150,17 +150,22 @@ class test_ucp_sockaddr : public ucp_test {
                 ucs::is_inet_addr(ifa->ifa_addr) &&
                 ucs::is_rdmacm_netdev(ifa->ifa_name))
             {
+                saddrs.push_back(ucs::sock_addr_storage());
                 status = ucs_sockaddr_sizeof(ifa->ifa_addr, &size);
                 ASSERT_UCS_OK(status);
-                m_test_addr.set_sock_addr(*ifa->ifa_addr, size);
-                m_test_addr.set_port(0); /* listen on any port then update */
-
-                freeifaddrs(ifaddrs);
-                return;
+                saddrs.back().set_sock_addr(*ifa->ifa_addr, size);
+                saddrs.back().set_port(0); /* listen on any port then update */
             }
         }
+
         freeifaddrs(ifaddrs);
-        UCS_TEST_SKIP_R("No interface for testing");
+
+        if (saddrs.size() > 0) {
+            /* randomize iface */
+            m_test_addr = saddrs[ucs::rand() % saddrs.size()];
```

Comment:
maybe just save pointers to "ifa" (and copy to m_test_addr before calling freeifaddrs)